### DISCUSSION TITLE
Access to an undefined property in custom package

### DISCUSSION THREAD

**Discussion by waelsjaber** [larastan/larastan#1255] [Q&A]
When I run `larastan` , I got an issue as below:

```
Access to an undefined property OwenIt\Auditing\Models\Audit::$auditable.
```

My code is:
```php

class AuditController extends Controller
{

    public function show(Request $request, Audit $model)
    {
       
        return  [
       'id' => $model->id,
       'auditable' => $model->auditable,
       'created_at' => $model->created_at->toDateString(),
        ];

    }
    
 }
```

How is the best way to solve this issue?

**Comment by canvural** [comment]
Hi,

Larastan scans your migrations or schema dumps to understand the model properties.

How is the `Audit` model defined? And are there migrations for it?

**Reply by waelsjaber** [reply]
Sure, this is how I  `use OwenIt\Auditing\Models\Audit` ,  Yes, I have migration.

**Comment by jakeydevs** [comment]
For anybody who is also getting this issue - I have fixed it by including a stub file with the Laravel Audit model and its properties annotated so that PHPstan can pick it up.

For reference, the files are as follows:

/stubs/Audit.stub.php
```
<?php

namespace OwenIt\Auditing\Models;

use Carbon\Carbon;

/**
 * @property string $auditable_type
 * @property Carbon $created_at
 * @property string $event
 * @property int $id
 * @property mixed $auditable
 */
class Audit
{
    //-- ..
}

```
To get PHPstan to read this, inside my phpstan.neon configuration file I have:

```
parameters:
    stubFiles:
        - stubs/Audit.stub.php
```
